client/active-game/game-server.js

```diff
@@ -1,80 +1,97 @@
 import log from '../logging/logger'
 import activeGameManager from './active-game-manager-instance'
-import { sendCommand, subscribeToCommands } from './game-command'
 import { mergeLocalSettings } from '../settings/action-creators'
 import { dispatch } from '../dispatch-registry'
 
 let lastLog = -1
 const logThrottle = 30000
-function authorize(req, cb) {
-  const origin = req.headers.origin
+function authorize(req) {
+  const origin = req.origin
   // We only accept connections from the game (or at the very least, things that can control their
   // origin, i.e. not browsers)
   if (origin !== 'BROODWARS') {
     if (Date.now() - lastLog > logThrottle) {
       lastLog = Date.now()
       log.warning('Blocked a connection from an untrusted origin: ' + origin)
     }
-    cb(null, false)
-    return
+    return false
   }
-  cb(null, true)
+  return true
 }
 
-function registerGameRoutes(nydus) {
-  async function getGameId(data, next) {
-    const id = data.get('client').conn.request.headers['x-game-id']
-    const newData = data.set('gameId', id)
-    await next(newData)
-  }
-
-  async function onSetupProgress(data, next) {
-    activeGameManager.handleSetupProgress(data.get('gameId'), data.get('body').status)
-  }
-
-  async function onStart(data, next) {
-    activeGameManager.handleGameStart(data.get('gameId'))
-  }
+class GameServer {
+  constructor(server) {
+    this.idToSocket = new Map()
+
+    activeGameManager.on('gameCommand', (id, command, payload) => {
+      log.verbose(`Sending game command to ${id}: ${command}`)
+      const socket = this.idToSocket.get(id)
+      if (socket) {
+        this.sendCommand(socket, command, payload)
+      } else {
+        // Is this an bad error or something that commonly occurs? Guessing that it's common.
+        log.verbose(`No game connection for ${id}`)
+      }
+    })
 
-  async function onEnd(data, next) {
-    const body = data.get('body')
-    activeGameManager.handleGameEnd(data.get('gameId'), body.results, body.time)
+    server.on('connection', (socket, request) => {
+      const gameIdIndex = request.rawHeaders.indexOf('x-game-id')
+      if (gameIdIndex !== -1) {
+        const gameId = request.rawHeaders[gameIdIndex + 1]
+        log.verbose('game websocket connected')
+        socket.on('close', () => {
+          log.verbose('game websocket disconnected')
+          this.idToSocket.delete(gameId)
+        })
+        socket.on('message', message => {
+          this.onMessage(gameId, message)
+        })
+        this.idToSocket.set(gameId, socket)
+        activeGameManager.handleGameConnected(gameId)
+      }
+    })
   }
 
-  async function onReplaySave(data, next) {
-    const { path } = data.get('body')
-    activeGameManager.handleReplaySave(data.get('gameId'), path)
+  sendCommand(socket, command, payload) {
```

Comment:
Maybe put underscore before the name of this method to indicate it should stay private? Otherwise, this method should probably be the one that receives the `gameId` and fetches socket based on that and checks if the socket is still open/valid.